src/devices/bus/ieee488/hp9133.cpp

```diff
@@ -0,0 +1,878 @@
+	virtual void ieee488_srq(int state) override;
+	virtual void ieee488_atn(int state) override;
+	virtual void ieee488_ren(int state) override;
+
+private:
+
+	static constexpr int IO1_R_BCS_N = 0x10;
+	static constexpr int IO1_R_BDRQ = 0x20;
+	static constexpr int IO1_R_HDINT = 0x40;
+	static constexpr int IO1_R_HDIND = 0x80;
+
+	static constexpr int IO2_R_FDCINT = 0x10;
+	static constexpr int IO2_R_FDKDCH_N = 0x20;
+	static constexpr int IO2_R_WRTFLTERR = 0x80;
+
+	static constexpr int IO3_R_FASTDON = 0x20;
+	static constexpr int IO3_R_SELTST = 0x40;
+	static constexpr int IO3_R_TEST = 0x80;
+
+	static constexpr int IO1_W_FLT_N = 0x01;
+	static constexpr int IO1_W_DS1 = 0x02;
+	static constexpr int IO1_W_BRDY = 0x40;
+	static constexpr int IO1_W_WDRESET_N = 0x80;
+
+	static constexpr int IO2_W_FDCRES_N = 0x01;
+	static constexpr int IO2_W_CLRIND_N = 0x04;
+	static constexpr int IO2_W_DKCHRES_N = 0x08;
+	static constexpr int IO2_W_FDHDLD_N = 0x10;
+	static constexpr int IO2_W_FDMOTON_N = 0x20;
+	static constexpr int IO2_W_FDHDSEL_N = 0x40;
+
+	static constexpr int IO3_W_INTSEL_MASK = 0x03;
+	static constexpr int IO3_W_INTENBL_N = 0x04;
+	static constexpr int IO3_W_FAST = 0x10;
+	static constexpr int IO3_W_DMAR_W_N = 0x20;
+	static constexpr int IO3_W_HPB_FDC_N = 0x40;
+	static constexpr int IO3_W_DMA = 0x80;
+
+	required_device<mc6809_device> m_cpu;
+	required_device<i8291a_device> m_gpib;
+	required_device<wd2010_device> m_hdc;
+	required_device<harddisk_image_device> m_harddisk;
+	required_device<wd2793_device> m_fdc;
+	required_device<floppy_connector> m_floppy;
+	required_ioport m_hpib_addr;
+	required_ioport m_selftest;
+	required_ioport m_selftesten;
+	required_ioport m_volcfg;
+	required_ioport m_model;
+
+	TIMER_CALLBACK_MEMBER(floppy_motor_timeout);
+	TIMER_CALLBACK_MEMBER(index_timer);
+	TIMER_CALLBACK_MEMBER(fast_timer);
+
+	void i8291a_eoi_w(int state);
+	void i8291a_dav_w(int state);
+	void i8291a_nrfd_w(int state);
+	void i8291a_ndac_w(int state);
+	void i8291a_ifc_w(int state);
+	void i8291a_srq_w(int state);
+	void i8291a_atn_w(int state);
+	void i8291a_ren_w(int state);
+	void i8291a_dio_w(uint8_t data);
+	void i8291a_int_w(int state);
+	void i8291a_drq_w(int state);
+	void fdc_intrq_w(int state);
+	void fdc_drq_w(int state);
+	void hdc_reg_w(offs_t offset, uint8_t data);
+	uint8_t hdc_reg_r(offs_t offset);
+	uint8_t i8291a_dio_r();
+	void io1_w(uint8_t data);
+	void io2_w(uint8_t data);
+	void io3_w(uint8_t data);
+	uint8_t io1_r(void);
+	uint8_t io2_r(void);
+	uint8_t io3_r(void);
+
+	void dma_ram_w(uint8_t data);
+	uint8_t dma_ram_r(offs_t offset);
+	void cpu_map(address_map &map);
+	void floppy_index_cb(floppy_image_device *floppy, int state);
+	void hdc_bcs_cb(int state);
+	void hdc_bcr_cb(int state);
+	void hdc_bdrq_cb(int state);
+	void hdc_intrq_cb(int state);
+	void hdc_wg_cb(int state);
+	void hdc_readwrite_sector(bool write);
+	void update_intsel(void);
+	void cpu_sync_ack(int state);
```

Comment:
Convention in MAME is not to put the `void` in the parentheses for functions with no arguments (use empty parentheses).

Reply:
Ok.